src/Preferences.cc

```diff
@@ -224,16 +224,15 @@ void Preferences::init() {
         initComboBox(this->comboBoxRotationY, Settings::Settings::inputRotateY);
         initComboBox(this->comboBoxRotationZ, Settings::Settings::inputRotateZ);
         initComboBox(this->comboBoxZoom, Settings::Settings::inputZoom);
-        initComboBox(this->comboBoxButton0, Settings::Settings::inputButton0);
-        initComboBox(this->comboBoxButton1, Settings::Settings::inputButton1);
-        initComboBox(this->comboBoxButton2, Settings::Settings::inputButton2);
-        initComboBox(this->comboBoxButton3, Settings::Settings::inputButton3);
-        initComboBox(this->comboBoxButton4, Settings::Settings::inputButton4);
-        initComboBox(this->comboBoxButton5, Settings::Settings::inputButton5);
-        initComboBox(this->comboBoxButton6, Settings::Settings::inputButton6);
-        initComboBox(this->comboBoxButton7, Settings::Settings::inputButton7);
-        initComboBox(this->comboBoxButton8, Settings::Settings::inputButton8);
-        initComboBox(this->comboBoxButton9, Settings::Settings::inputButton9);
+
+        for (int i = 0; i < 10; i++ ){ 
+			std::string s = std::to_string(i);
+            QComboBox* box = this->centralwidget->findChild<QComboBox *>(QString::fromStdString("comboBoxButton"+s));
+			Settings::SettingsEntry* ent = Settings::Settings::inst()->getSettingEntryByName("button" +s );
+			if(box != 0 && ent != nullptr){
```

Comment:
s/0/nullptr/ .. or simply `if (bot && ent)`

Reply:
Thank you. I am coming from a Javascript background - that left some paranoia when checking for zero and null.
Change commited.